### PR TITLE
Support NEW_PASSWORD_REQUIRED challenge

### PR DESCRIPTION
**Pull request by klesh**
### Summary

For security reasons, a new landing tenant should be challenged to set a new password for the account.
This PR supports NEW_PASSWORD_REQUIRED challenge during the `login` procedure.

### Screenshots
cognito-errmsg-passthrough
prompt-new-password
set-new-password
log-on-successfully

### CODE REVIEW

File: env.example [hunk 1]
```diff
@@ -35,6 +35,8 @@ FORCE_MIGRATION=false
 # Lake TAP API
 TAP_PROPERTIES_DIR=
 
+DISABLED_REMOTE_PLUGINS=
```
Comment (abeizn) [inline env.example:38]: skip azuredevops by default？

File: backend/server/services/auth/cognito.go [hunk 2]
```diff
@@ -189,3 +203,52 @@ type Jwks struct {
...
+			ExpiresIn:    response.AuthenticationResult.ExpiresIn,
+			IdToken:      response.AuthenticationResult.IdToken,
+			RefreshToken: response.AuthenticationResult.RefreshToken,
+			TokenType:    response.AuthenticationResult.TokenType,
+		}
+	}
+	return loginRes, nil
+}
+
+// func (cgt *AwsCognitorProvider) ChangePassword(ctx *gin.Context, oldPassword, newPassword string) errors.Error {
```
Comment (abeizn) [inline backend/server/services/auth/cognito.go:239]: Is this follow-up still useful?

File: backend/server/api/login/login.go [hunk 3]
```diff
@@ -18,56 +18,64 @@ limitations under the License.
...
-}
-type ChallengeParameters struct {
-}
-
 // @Summary post login
 // @Description post login
 // @Tags framework/login
 // @Accept application/json
-// @Param blueprint body LoginRequest true "json"
+// @Param blueprint body aut.LoginRequest true "json"
```
Comment (abeizn) [inline backend/server/api/login/login.go:34]: auth?

File: env.example (hunk 1 above)
Comment (klesh) [reply env.example:38]: why? this is enable by default unless you specify the EnvVar to be `true`

File: backend/server/api/login/login.go (hunk 3 above)
Comment (hezyin) [reply backend/server/api/login/login.go]: @klesh Could you please clarify what blueprint mean in this annotation?

Also the `aut` needs to be corrected.

**Review by hezyin — COMMENTED** [review]
Overall looks good, please see my comments for some quick questions.

File: backend/server/services/auth/cognito.go [hunk 4]
```diff
@@ -26,129 +26,133 @@ import (
...
 )
 
-var (
-	//jwksCache is a cache of the fetched JWKS
-	jwksCache Jwks
-	//jwksCacheMx is a mutex to lock the jwksCache
-	jwksCacheMx sync.Mutex
-	logger      = logruslog.Global.Nested("auth")
-)
+type AwsCognitorProvider struct {
```
[the remark this answers is not in the document]
Comment (klesh) [reply backend/server/services/auth/cognito.go:40]: typo 🥲

File: backend/server/services/auth/auth.go [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+type AuthenticationResult struct {
+	AccessToken  *string `json:"accessToken" type:"string" sensitive:"true"`
+	ExpiresIn    *int64  `json:"expiresIn" type:"integer"`
+	IdToken      *string `json:"idToken" type:"string" sensitive:"true"`
+	RefreshToken *string `json:"refreshToken" type:"string" sensitive:"true"`
+	TokenType    *string `json:"tokenType" type:"string"`
+}
+
+type ChallengeParameters struct {
```
[the remark this answers is not in the document]
Comment (klesh) [reply backend/server/services/auth/auth.go:52]: Maybe keep it for a while, it is for other challenges like MFA, I will add comment to the struct.

File: backend/server/api/login/login.go (hunk 3 above)
Comment (klesh) [reply backend/server/api/login/login.go]: It is the name of the parameter from the previous code which I believe was copied from the `api/blueprint` folder and should be renamed. 
I didn't notice the spelling

File: backend/server/services/auth/auth.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (klesh) [reply backend/server/services/auth/auth.go:52]: Oh, you are correct, it was replaced by a `map` already, we don't need it anymore.

**Pull request merged**